openraft/src/core/mod.rs

```diff
@@ -764,6 +798,23 @@ struct LeaderState<'a, C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStora
     pub(super) awaiting_committed: Vec<ClientRequestEntry<C>>,
 }
 
+impl<'a, C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> MetricsReporter<C>
+    for LeaderState<'a, C, N, S>
+{
+    fn get_update_metrics_option(
+        &self,
+        option: Option<UpdateMetricsOption>,
+    ) -> Option<Update<Option<Versioned<LeaderMetrics<C::NodeId>>>>> {
+        match option {
+            Some(op) => match op {
+                UpdateMetricsOption::AsIs => Some(Update::AsIs),
+                UpdateMetricsOption::Update => Some(Update::Update(Some(self.leader_metrics.clone()))),
+            },
+            None => None,
+        }
```

Comment:
To simplify these two matching, you can just add another variant, such as `NoUpdate`:
```rust
pub enum UpdateMetricsOption {
    NoUpdate,
    Update,
    AsIs,
}
```